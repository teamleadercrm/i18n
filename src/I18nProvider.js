import React, { PureComponent } from 'react';
import { createIntl, createIntlCache, RawIntlProvider } from 'react-intl';

import { getTranslationPath } from './utils';

export const FALLBACK_LANGUAGE = 'en';

let translate = undefined;

class I18nProvider extends PureComponent {
  state = {
    locale: this.props.locale || FALLBACK_LANGUAGE,
    messages: {},
    isLoading: true,
  };

  async componentDidMount() {
    try {
      const response = await fetch(getTranslationPath(this.props.path, this.props.locale));
      const translations = await response.json();
      this.setState({
        isLoading: false,
        messages: translations,
      });
    } catch (error) {}
  }

  getIntl() {
    const cache = createIntlCache();

    const intl = createIntl(
      {
        locale: this.state.locale,
        messages: this.state.messages,
      },
      cache,
    );

    translate = (id, values) => {
      const { namespace } = this.props;
      const namespacedId = namespace ? [namespace, id].join('.') : id;

      return intl.formatMessage({ id: namespacedId }, values);
    };

    return intl;
  }

  render() {
    if (this.state.isLoading) {
      return null;
    }

    return <RawIntlProvider value={this.getIntl()}>{this.props.children}</RawIntlProvider>;
  }
}

export default I18nProvider;
export { translate };
