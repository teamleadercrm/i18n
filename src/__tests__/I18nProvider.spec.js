import React from 'react';
import { mount } from 'enzyme';
import I18nProvider from '../I18nProvider';

describe('I18nProvider', () => {
  it('renders the its children', () => {
    const rendered = mount(
      <I18nProvider>
        <div>Hello World</div>
      </I18nProvider>,
    );
    expect(rendered.children().length).toBe(1);
    expect(rendered).toMatchSnapshot();
  });
});
