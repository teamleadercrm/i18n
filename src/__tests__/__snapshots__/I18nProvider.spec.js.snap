// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`I18nProvider renders the its children 1`] = `ReactWrapper {}`;
